BUG: Update CTKAppLauncher from 0.1.30 to 31 to support Qt5 reserved arguments (#6351)
Slicer/Slicer: Docs/user_guide/settings.md

- `SLICER_EXTENSIONS_MANAGER_SERVER_URL`: URL of the extensions manager backend with the `/api` path. Default value is retrieved from the settings using the key `Extensions/ServerUrl`.
- `SLICER_EXTENSIONS_MANAGER_FRONTEND_SERVER_URL`: URL of the extension manager frontend displaying the web page. Default value is retrieved from the settings using the key `Extensions/FrontendServerUrl`.
- `SLICER_EXTENSIONS_MANAGER_SERVER_API`: Supported values are `Midas_v1` and `Girder_v1`. Default value is hard-coded to `Girder_v1`.

### Qt built-in command-line options

> Slicer application supports specific command-line options that allow modifying the way Qt will interact with the windowing system.

Reviewer: ```suggestion
Slicer application accepts standard Qt command-line arguments that specify how Qt interacts with the windowing system.
```

Examples of options:

> - `-widgetcount`, prints debug message at the end about number of widgets left undestroyed and maximum number of widgets existed at the same time.

Reviewer: Are there any better examples of options that could be actually useful? For example, controlling DPI awareness could be useful. Or setting the window geometry or display number.

- `-reverse`, sets the application's layout direction to `Qt::RightToLeft`.

To learn about the supported options:
- https://doc.qt.io/qt-5/qapplication.html#QApplication
- https://doc.qt.io/qt-5/qguiapplication.html#supported-command-line-options

:::{note}

> Since the Slicer launcher is itself a Qt application and the Qt built-in command-line options are expected to be passed to

Author: I clarified further

- `SLICER_EXTENSIONS_MANAGER_SERVER_URL`: URL of the extensions manager backend with the `/api` path. Default value is retrieved from the settings using the key `Extensions/ServerUrl`.
- `SLICER_EXTENSIONS_MANAGER_FRONTEND_SERVER_URL`: URL of the extension manager frontend displaying the web page. Default value is retrieved from the settings using the key `Extensions/FrontendServerUrl`.
- `SLICER_EXTENSIONS_MANAGER_SERVER_API`: Supported values are `Midas_v1` and `Girder_v1`. Default value is hard-coded to `Girder_v1`.

### Qt built-in command-line options

Slicer application accepts standard Qt command-line arguments that specify how Qt interacts with the windowing system.

Examples of options:
- `-qwindowgeometry geometry`, specifies window geometry for the main window using the X11-syntax. For example: `-qwindowgeometry 100x100+50+50`.
- `-display hostname:screen_number`, switches displays on X11 and overrides the `DISPLAY` environment variable.

> - `-platform windows:dpiawareness=[0|1|2]`, sets the [DPI awareness](https://doc.qt.io/qt-5/highdpi.html#microsoft-windows) on Windows.

Author: I will have to confirm that this syntax works as expected. Will get back to this a few hours ...

Author: > I will have to confirm that this syntax works as expected. Will get back to this a few hours ...

:heavy_check_mark: See https://github.com/commontk/AppLauncher/pull/126

Reviewer: These are good examples. Hopefully they work!
